better_profanity/varying_string.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+
+class VaryingString:
+    """Represents a string with varying character representations."""
+
+    def __init__(self, string, char_map={}, variant_thres=10000):
+        """
+        Args:
+            string (str): String to generate variants of.
+            char_mappings (dict): Maps characters to substitute characters.
+            variant_thres (int): Maximum number of variants to store in a batch.
```

Comment:
It seems `variant_thres` is not used ? And do we need to use it when we have the `MAX_PATTERNS` constant in #16

Reply:
The PR looks good to me. I will merge after `variant_thres` is added. What do you think about renaming it to `max_variants` ?

Reply:
Oh my bad, `variant_thres` isn't used anymore. It was used for a less-efficient method I experimented with and I forgot to remove it.

`MAX_PATTERNS` won't be used anymore. The character-by-character string comparison methods doesn't actually construct any of the variants so it's not relevant anymore.

Reply:
It's removed now. Should be ready to pull.